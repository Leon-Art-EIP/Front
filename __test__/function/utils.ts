import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
