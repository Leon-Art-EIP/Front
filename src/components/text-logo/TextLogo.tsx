export default function TextLogo() {
  return (
    <>
      <span className="text-[#E11C0A]">Leon</span>
      <span className="text-[#000000]">&apos;Art</span>
    </>
  );
}
