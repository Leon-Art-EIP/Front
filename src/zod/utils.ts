export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
