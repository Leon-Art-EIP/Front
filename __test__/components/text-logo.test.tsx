import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import TextLogo from "../../src/components/text-logo/TextLogo";

test("TextLogo snapshot", () => {
  testSnapshot(<TextLogo />);
});
