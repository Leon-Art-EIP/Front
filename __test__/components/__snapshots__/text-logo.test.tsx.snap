// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextLogo snapshot 1`] = `
[
  <span
    className="text-primary"
  >
    Leon
  </span>,
  <span
    className="text-tertiary"
  >
    'Art
  </span>,
]
`;
